src/Tribe/Meta/Chunker.php

```diff
@@ -0,0 +1,811 @@
+		$ordered_chunks = array();
+		foreach ( $chunks as $chunk ) {
+			preg_match( '/(\\d+)' . preg_quote( $this->chunk_separator ) . '(.*)/', $chunk, $matches );
+			$ordered_chunks[ $matches[1] ] = $matches[2];
+		}
+		ksort( $ordered_chunks );
+
+		return implode( '', array_values( $ordered_chunks ) );
+	}
+
+	/**
+	 * Returns the chunks stored in the database for a post ID and meta key couple.
+	 *
+	 * The chunks are returned as they are with prefix.
+	 *
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 *
+	 * @return array|mixed
+	 */
+	public function get_chunks_for( $object_id, $meta_key ) {
+		$key = $this->get_key( $object_id, $meta_key );
+
+		if ( ! empty( $this->chunks_cache[ $key ] ) ) {
+			return $this->chunks_cache[ $key ];
+		}
+
+		/** @var wpdb $wpdb */
+		global $wpdb;
+
+		$chunk_meta_key = $this->get_chunk_meta_key( $meta_key );
+
+		$meta_ids = $wpdb->get_col( $wpdb->prepare( "SELECT meta_id FROM {$wpdb->postmeta}
+			WHERE post_id = %d
+			AND meta_key = %s",
+			$object_id, $chunk_meta_key
+		) );
+
+		$meta_values = array();
+		foreach ( $meta_ids as $meta_id ) {
+			$query = $wpdb->prepare( "SELECT meta_value FROM {$wpdb->postmeta} WHERE meta_id = %d", $meta_id );
+			$meta_values[] = $wpdb->get_var( $query );
+		}
+
+		if ( ! empty( $meta_values ) ) {
+			$this->chunks_cache[ $this->get_key( $object_id, $meta_key ) ] = $meta_values;
+		} else {
+			$this->chunks_cache[ $this->get_key( $object_id, $meta_key ) ] = null;
+		}
+
+		return $meta_values;
+	}
+
+	/**
+	 * Resets a post ID and meta key couple cache.
+	 *
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 */
+	protected function cache_delete( $object_id, $meta_key ) {
+		$key = $this->get_key( $object_id, $meta_key );
+		if ( isset( $this->chunks_cache[ $key ] ) ) {
+			$this->chunks_cache[ $key ] = null;
+		}
+	}
+
+	/**
+	 * Filters the delete operations.
+	 *
+	 * @param mixed  $check
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 *
+	 * @return bool
+	 *
+	 * @see delete_metadata()
+	 */
+	public function filter_delete_metadata( $check, $object_id, $meta_key ) {
+		if ( ! $this->applies( $object_id, $meta_key ) ) {
+			return $check;
+		}
+
+		$has_chunked_meta = $this->is_chunked( $object_id, $meta_key );
+		if ( ! $has_chunked_meta ) {
+			return $check;
+		}
+		$this->cache_delete( $object_id, $meta_key );
+		$this->delete_chunks( $object_id, $meta_key );
+
+		return true;
+	}
+
+	/**
+	 * Whether a post ID and meta key couple has chunked meta or not.
+	 *
+	 * @param int    $object_id
+	 * @param string $meta_key
+	 * @param bool   $check_db Do verify the chunking state on the database.
+	 *
+	 * @return mixed
+	 */
+	public function is_chunked( $object_id, $meta_key, $check_db = false ) {
+		$key = $this->get_key( $object_id, $meta_key );
+		$chunked_in_cache = array_key_exists( $key, $this->chunks_cache ) && is_array( $this->chunks_cache[ $key ] );
+
+		return false === $check_db ?
+			$chunked_in_cache
```

Comment:
Move the `?` operator from the previous line to the start of this line? Would look a little cleaner and be more consistent with other code / with the following line.